migrate/20230919_add_vm_pool.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  change do
+    create_table(:vm_pool) do
+      column :id, :uuid, primary_key: true, default: nil
+      column :size, :integer, null: false
+      column :vm_size, :text, null: false
+      column :boot_image, :text, null: false
+      column :location, :text, null: false
+    end
```

Comment:
I gave some thought about this proposal. In theory I like it. It would simplify the overall design and shorten it. If you think about it, identifying a VM from a pool gets even shorter in the first look. But that's the first look.

Thinking about it more, I start to see certain drawbacks. For example;
1. Creating repository specific pools would not be much possible if we don't start to store this information in VM model.
2. Creating a customer specific pool would not be possible because of the same reason. 
3. Filtering different vm_pools start to become an issue especially if we have pools with overlapping rules. Picking the VM from the right pool gets tricky. We would try to get the VM with the highest number of matching rules but that would complicate the algorithm in the pick_vm function.

I understand that the proposed design has some advantages, but to be honest, I quite like the structured, clean form of this PR. 

One of my reasons is (not really a good reason maybe), but this is the most basic, the first thing I think when I think about a VM pool implementation. It's easy to grasp the idea, it's simple, it's straightforward to iterate over it. Keeping the pool parameters separate has it's own advantages.

With my sincere respect to @fdr 's terrible ideas :)

Reply:
Looking into it more, destroying a specific pool of VMs get kind of messy and error prone with filtering getting difficult to use.

Reply:
If we need to implement dynamic size pooling, we'll require a prog to manage it. The proposal works well for a static pool size.

Reply:
yeah, thinking about it more, I start to see more drawbacks, at this point the new proposal doesn't feel right to me.

Reply:
I suppose keeping in the spirit of terrible ideas (I am flattered that this has become A Thing), the root general idea at issue is the duplication. The thing about terrible ideas is sometimes you can wind your way towards an idea that is not as terrible, but the terrible idea is usable as a foil (literary sense) in that process.

What comes to mind for this in particular, I suppose a variant of this would be to have some kind of Vm Template concept that has a degenerate use in being joined around in multiple ways, as Sshable can be joined with either Vms or VmHosts or any number of things, or no things (though it'd be a bit odd).

It's quite possible that any number of unforeseen complications could undermine such a re-factoring (in the most strict sense, since a common factor is extruded), but...it's what comes to mind for me first.

Reply:
I'm glad that we had this discussion. It seems the terrible idea is indeed terrible :smile:, but at least the process itself is useful on its own. Thanks @furkansahin for implementing the proposal. Low line number count is definitely enticing, but it has drawbacks as discussed in this thread. I'm fine with moving ahead with the current solution.

Vm Templates would be interesting. Though it would require bigger refactoring. If we had a 3rd place where we would use such join table, cost of refactoring would be justified IMO.

Reply:
Me too! Thanks team 😃